shuler_lab_to_nwb/fiberphotometry_datainterface/tools.py

```diff
@@ -0,0 +1,117 @@
+from typing import Optional
+import pandas as pd
+from hdmf.backends.hdf5 import H5DataIO
+from hdmf.common import DynamicTableRegion
+from ndx_photometry import FibersTable, FiberPhotometry, ExcitationSourcesTable, PhotodetectorsTable, FluorophoresTable
+from pynwb import NWBFile
+from pynwb.ophys import RoiResponseSeries
+
+
+def read_fp_file(file_path: str) -> pd.DataFrame:
+    """Reads a fiber photometry file (csv) and returns a pandas DataFrame."""
+    df = pd.read_csv(
+        file_path,
+        sep=" ",
+        header=None,
+        names=["timestamp", "unknown", "GFP_first_branch", "RFP_first_branch", "GFP_second_branch", "RFP_second_branch"],
+        index_col=False
+    )
+    df = df.drop(columns=["unknown"])
+    df["timestamp"] = df["timestamp"] / 1000
+
+    # Separate signal and isosbestic dataframes using the mask: True for signal rows, False for isosbestic rows
+    mask = df.index % 2 == 0
+    df_signal = df[mask]
+    df_isosbestic = df[~mask]
+    df_signal = df_signal.reset_index(drop=True)
+    df_isosbestic = df_isosbestic.reset_index(drop=True)
+
+    # Rename isosbestic columns to include _isosbestic
+    df_isosbestic = df_isosbestic.rename(columns={
+        "GFP_first_branch": "GFP_first_branch_isosbestic", 
+        "RFP_first_branch": "RFP_first_branch_isosbestic", 
+        "GFP_second_branch": "GFP_second_branch_isosbestic", 
+        "RFP_second_branch": "RFP_second_branch_isosbestic"
+    })
+    return df_signal, df_isosbestic
+
+
+def add_photometry(
```

Comment:
Thinking about how this would look in NeuroConv; for modularity, would suggest breaking each major step into `add_{table type}_table` and finally `add_photometry_response_series`

Reply:
@CodyCBakerPhD 
yeah, I've been thinking if this couldn't be the basis for a `BaseFiberPhotometryDataInterface` class within neuroconv, what do you think?
I can include this code modularized like this over there, then just import here.

Reply:
Not a class; these would be the basis of `neuroconv.tools.photometry` which would then be _used_ by interfaces, but the interfaces are more about how to read from the source format (during `__init__`) whereas the tools are focused all on in-memory operations once you have arrays and dataframes and such